_posts/2020-10-16-kaitai-struct-v0.9-released.md

```diff
@@ -13,41 +13,43 @@ categories: news
 After a lot of time and effort, Kaitai project is happy to announce
 release of new major version of Kaitai Struct, declarative markup
 language to describe various binary data structures — binary file
-formats, network stream packets, etc.
+formats, network packets, etc.
 
 The basic idea of Kaitai Struct is that a particular format can be
-described using Kaitai Struct language (in a `.ksy` file), which then
-can be compiled using `kaitai-struct-compiler` into source files in
-one of the supported programming languages. These modules will include
-a generated code for a parser that can read described data structure
-from a file / stream and provide access to its contents in a nice,
+described using Kaitai Struct language (in a `.ksy` file, that stands
+for Kaitai Struct YAML), which then can be compiled using
+`kaitai-struct-compiler` into source files in one of the supported
+programming languages. These modules will include a generated
+code for a parser that can read described data structure from a
+file / stream and provide access to its contents in a nice,
 easy-to-comprehend API.
 
 With the previous 0.8 release, Kaitai project celebrated 1000 stars on GitHub,
 and until 0.9 version, it has collected more than [2000 stars](https://github.com/kaitai-io/kaitai_struct/stargazers).
 Thank you all for your support!
 
 This version introduces C++11 target (which uses smart pointers),
-several handy features (like `valid`ations and little-endian bit integers),
-fixes a lot of bugs and includes quite a few infrastructure improvements.
+several handy features (like `valid`ations and little-endian "bit-sized
+types"), fixes a lot of bugs and includes quite a few infrastructure
```

Comment:
I would keep the word _integers_, because _type_ is too broad in my opinion:

```suggestion
several handy features (like `valid`ations and little-endian "bit-sized
types"), fixes a lot of bugs and includes quite a few infrastructure
```

Reply:
Because endianness issue is not about bit-sized integers alone, it is about both bit- and byte-sized integers within types which fields (which can have non-integer type) can occupy non-integer count of bytes. Endeanness is here because of remapping. The common way to deal with bit-sized stuff is using bitwise logical operations, since it is the only mean  of doing that on most of CPUs (maybe except some Intel ones with integrated FPGAs), and it is proficient to use CPU-native remapping, little-endian (no designers of modern systems are mad enough to use BE as native byte order). Such remapping usually deals with chunks longer than bytes, and it is position of numbers and structs relative to the chunks boundaries defines the way pieces of chunks are cut, reordered and joined. So, IMHO, endianness for bit-sized integers alone makes no sense as it is at all (how to cut `b9le`, `0b11111111_1`, , `0b1_11111111`, or maybe `0b11111_1111`?), what makes sense is the endianness of their container that has the bit-size of multiple of chunk length and aligned to chunks boundaries and where the field is within the container.